src/librustc_mir/transform/const_prop.rs

```diff
@@ -852,20 +856,34 @@ impl<'mir, 'tcx> MutVisitor<'tcx> for ConstPropagator<'mir, 'tcx> {
                     if let Some(()) = self.const_prop(rval, place_layout, source_info, place) {
                         if can_const_prop == ConstPropMode::FullConstProp
                             || can_const_prop == ConstPropMode::OnlyPropagateInto
+                            || can_const_prop == ConstPropMode::OnlyInsideOwnBlock
                         {
+                            // This will return None for Locals that are from other blocks,
+                            // so it should be okay to propagate from here on down.
                             if let Some(value) = self.get_const(local) {
                                 if self.should_const_prop(value) {
                                     trace!("replacing {:?} with {:?}", rval, value);
                                     self.replace_with_const(rval, value, statement.source_info);
-
                                     if can_const_prop == ConstPropMode::FullConstProp {
+                                        // FIXME(felix91gr): is this trace accurate??
+                                        // See: next FIXME
                                         trace!("propagated into {:?}", local);
                                     }
                                 }
+                                if can_const_prop == ConstPropMode::OnlyInsideOwnBlock {
+                                    trace!(
+                                        "found local restricted to its block. Will remove it from const-prop after block is finished. Local: {:?}",
+                                        local
+                                    );
+                                    self.locals_of_current_block.insert(local);
+                                }
                             }
                         }
                     }
                     if self.can_const_prop[local] != ConstPropMode::FullConstProp {
+                        // FIXME(felix91gr): is this trace accurate??
```

Comment:
Yeah, this should change to `self.can_const_prop[local] == ConstPropMode::OnlyPropagateInto self.can_const_prop[local] == ConstPropMode::NoPropagation`.

Reply:
Nice! That seems to have done it, now the mir-opt tests are all in order. Let's see if we're having failing tests on the rest of the suite now :crossed_fingers: